app/models/group.rb

```diff
@@ -124,6 +126,15 @@ def add_leader_to_members
     members << leader unless members.include? leader
   end
 
+  # rubocop:disable RaiseArgs
+  def check_full_membership_validity
```

Comment:
Ohh rather than raising an exception? Maybe, let's chat about it offline and figure it out from there.